mobile/src/main/java/ch/epfl/sdp/mobile/ui/prepare_game/ColorChoiceTabBar.kt

```diff
@@ -0,0 +1,94 @@
+package ch.epfl.sdp.mobile.ui.prepare_game
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.selection.selectable
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.graphics.*
+import androidx.compose.ui.graphics.drawscope.Stroke
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.*
+
+enum class TabColors(val backgroundColor: Color, val textColor: Color, val borderColor: Color) {
+  SelectedColors(PawniesColors.Green800, PawniesColors.Orange200, PawniesColors.Orange200),
+  UnselectedColors(PawniesColors.Green100, PawniesColors.Green800, PawniesColors.Green100)
+}
```

Comment:
The compose-like way of doing this would be as follows :


```kotlin
interface ColorChoiceColors {
  @Composable
  fun background(selected: Boolean): State<Color>

  @Composable
  fun contentColor(selected: Boolean): State<Color>

  // Add your own fields etc.
}

private object DefaultColorChoiceColors {
  @Composable override fun background(selected: Boolean) = TODO()
  @Composable override fun contentColor(selected: Boolean) = TODO()
  // ...
}

// Which can then be passed to the composable function.
@Composable
fun ColorChoiceItem(
  selected: Boolean,
  modifier: Modifier = Modifier,
  colors: ColorChoiceColors = DefaultColorChoiceColors,
) {
  val color by colors.background(selected) // etc.
}

```

This is for instance exactly what the `Button` composable is doing ([source](https://github.com/androidx/androidx/blob/androidx-main/compose/material/material/src/commonMain/kotlin/androidx/compose/material/Button.kt#L261-L286)). I suggest you look at its API and try to mimic some of practices you observe in the standard Material components.

Reply:
Good point 👍 Thanks I will try to follow the guidelines.